fe-games/src/components/FullReview.jsx

```diff
@@ -1,7 +1,33 @@
+import { useState } from "react";
+import { useParams, Link } from "react-router-dom";
+
 const FullReview = () => {
+    const [review, setReview] = useState({});
+    const [comments, setComments] = useState([]);
+    const { review_id } = useParams();
+    const date = new Date(review.created_at);
+
+    fetch(`https://michaels-games.onrender.com/api/reviews/${review_id}`)
+    .then(response => response.json())
+    .then(data => {
+        setReview(data.review);
+    })
+
+    fetch(`https://michaels-games.onrender.com/api/reviews/${review_id}/comments`)
+    .then(response => response.json())
+    .then(data => {
+        setComments(data.comments);
+    })
```

Comment:
These fetch requests are happening on every render of the component. This will cause and infinite loop or re-renders and basically attempt to ddos your server (which is probably why it's being slow 😅 ). 

These are side-effects which should be ran inside of `useEffect` hooks with dependencies to control when they run.

Reply:
ah of course! I knew I was doing something wrong